flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBListState.java

```diff
@@ -237,4 +249,20 @@ public void addAll(List<V> values) throws Exception {
 
 		return keySerializationStream.toByteArray();
 	}
+
+	@Override
+	public List<V> getInternal() {
+		Iterable<V> list = get();
+		if (list == null) {
+			return null;
+		}
+		List<V> collected = new ArrayList<>();
```

Comment:
Maybe we could give this a safe init size, e.g (4).

Reply:
We could currently also safe the whole repacking if we change the signature of `Iterable<V> get()` in this class to return ``List<V>``.
However, I think in the long run it might be worth considering to have this class be based on `Iterable` instead of `List` because we essentially only use iterable semantics. @aljoscha what do you think?